src/proof/proof-service.ts

```diff
@@ -150,13 +173,39 @@ export class ProofService implements IProofService {
     this._prover = opts?.prover ?? new NativeProver(_circuitStorage);
     this._ldOptions = { ...opts, documentLoader: opts?.documentLoader ?? cacheLoader(opts) };
     this._inputsGenerator = new InputGenerator(_identityWallet, _credentialWallet, _stateStorage);
+    this._pubSignalsVerifier = new PubSignalsVerifier(
+      opts?.documentLoader ?? cacheLoader(opts),
+      _stateStorage
+    );
   }
 
   /** {@inheritdoc IProofService.verifyProof} */
   async verifyProof(zkp: ZKProof, circuitId: CircuitId): Promise<boolean> {
     return this._prover.verify(zkp, circuitId);
   }
 
+  /** {@inheritdoc IProofService.verify} */
+  async verifyZKPResponse(
+    proofResp: ZeroKnowledgeProofResponse,
+    opts: ProofVerifyOpts
+  ): Promise<BaseConfig> {
+    const proofValid = await this._prover.verify(proofResp, proofResp.circuitId);
+    if (!proofValid) {
+      throw Error(`Proof with circuit id ${proofResp.circuitId} and request id ${proofResp.id} is not valid`);
+    }
+
+    const verifyContext: VerifyContext = {
+      pubSignals: proofResp.pub_signals,
+      query: opts.query,
+      verifiablePresentation: proofResp.vp as JSON,
+      sender: opts.sender,
+      challenge: BigInt(proofResp.id),
+      opts: opts.opts,
+      params: opts.params
+    };
+    return this._pubSignalsVerifier.verify(proofResp.circuitId, verifyContext);
```

Comment:
let's return custom struct, like in solidity
VerificationResultMetadata and add single field for now linkId , and it will be assigned here.